core/app/components/pubs/PubEditor/actions.ts

```diff
@@ -145,18 +145,19 @@ export const _updatePub = async ({
 		const existingPubFieldValues = await autoCache(
 			trx
 				.selectFrom("pub_fields")
-				.leftJoin("pub_values", "pub_values.fieldId", "pub_fields.id")
+				.leftJoin("pub_values", (join) =>
+					join
+						.onRef("pub_values.fieldId", "=", "pub_fields.id")
+						.on("pub_values.pubId", "=", pubId)
+				)
```

Comment:
This query is intended to select field ids, field names, and field schemaNames for all of the values submitted so they can be validated. It has a secondary purpose of selecting pub_value ids for each of these values if they already exist. 
Before this change it was joining the values table without narrowing the results by the current pubId, then using distinctOn to filter out the extra rows.

Reply:
Also since we're already doing this query before the updates, arguably we should be selecting the values here too and using that information to avoid updating rows when the value doesn't change!